### PR TITLE
chore: todo message cleanup

### PR DESCRIPTION
**Pull request by nataliepina**
This removes TODO messages that are stale,
do not have a related task, or not necessary.

# Description

Prior to these changes we have around 16 TODO messages throughout the application. 
I've noticed that many of these are pretty old or don't have the proper context anymore. 
Some of these we will probably not get around to doing as well, therefore I think we can safely remove them. 

This will reduce the TODO maintenance burden and improve the professionalism of the code. 
Over time, codebases evolve, and priorities change. Old TODOs may no longer be relevant to the current state of the project. Removing them keeps the codebase clean and ensures that we can focus on addressing current tasks and issues. Accumulating a large number of unresolved TODOs can clutter the code. 

I have addressed the remaining todos in the work done [in this pull request](https://github.com/d2iq/ui-kit/pull/998). 
So after this one, we will have 0 TODO messages in UI Kit! 

## Testing
Review the TODOS to be removed. 

## Checklist

- [ ] This PR is associated with a JIRA and it is mentioned in the commit message footer ("Closes …")
- [ ] Significant changes have been tested downstream to avoid breaking changes
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [x] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

File: packages/appChrome/style.ts [hunk 1]
```diff
@@ -88,8 +88,6 @@ export const sidebarContainer = (theme: AppChromeTheme, isOpen?: boolean) => {
   `;
 };
 
-// TODO: replace animation duration/easing with design tokens
-// once design has agreed on animation
 export const sidebarAnimator = css`
```
Comment (nataliepina) [inline packages/appChrome/style.ts:91]: We have no plans for animation design tokens. I think we can consider this in the future, but for now it is not relevant.

File: packages/button/components/ButtonBase.tsx [hunk 2]
```diff
@@ -54,12 +54,10 @@ export interface ButtonProps extends LinkProps {
   /**
    * The icon that appears before the button text
    */
-  // TODO: only accept IconShapes when we make a big breaking change
   iconStart?: IconShapes | React.ReactElement<HTMLElement>;
```
Comment (nataliepina) [inline packages/button/components/ButtonBase.tsx:57]: I created a JIRA to follow up and track these two - https://d2iq.atlassian.net/browse/D2IQ-99153

File: packages/shared/styles/color.ts [hunk 3]
```diff
@@ -53,18 +53,12 @@ const getLuminance = (luminanceVal1: number, luminanceVal2: number) => {
   return (l1 + 0.05) / (l2 + 0.05);
 };
 
-// TODO: when/if we start using rgb/rgba colors in
-// design-tokens, write getHexContrast function that can get
-// rgb/rgba contrast as well
```
Comment (nataliepina) [inline packages/shared/styles/color.ts:58]: These two are 4 years old. I do not see us using rgb/rgba colors in the foreseeable future. I don't think this is a great TODO to leave for future devs without context or a task.

File: packages/shared/styles/normalize.ts [hunk 4]
```diff
@@ -7,7 +7,6 @@ import {
   themeBgPrimary
 } from "../../design-tokens/build/js/designTokens";
 
-// TODO: follow-up with Jira ticket DCOS-62772 (a more thorough set of normalize styles)
```
Comment (nataliepina) [inline packages/shared/styles/normalize.ts:10]: This JIRA has been resolved.

File: packages/shared/styles/styleUtils/typography/color.ts [hunk 5]
```diff
@@ -6,9 +6,6 @@ import {
   themeTextColorSecondaryInverted
 } from "../../../../design-tokens/build/js/designTokens";
 
-// TODO: when we do the design tokens lib,
-// generate a type for _only_ our colors
-// instead of just accepting any string
 export const tintText = (color: React.CSSProperties["color"]) => css`
```
Comment (nataliepina) [inline packages/shared/styles/styleUtils/typography/color.ts:9]: I don't have the full proper context to see the need for this one, and it is 5 years old. I feel we can safely remove.

File: packages/tablev2/tablev2.test.tsx [hunk 6]
```diff
@@ -229,9 +229,6 @@ describe("Table v2", () => {
         expect.objectContaining({ sortBy: "name", order: "desc" })
       );
     });
-    // TODO: figure out how we could simulate resizing
-    /* eslint-disable @typescript-eslint/no-empty-function */
-    it.skip("calls onStateChange when a column is resized", () => {});
   });
```
Comment (nataliepina) [inline packages/tablev2/tablev2.test.tsx:232]: We are skipping this test 🤷🏻‍♀️

File: packages/themes/stories/UIKitThemeProvider.stories.tsx [hunk 7]
```diff
@@ -1,5 +1,3 @@
-// TODO fix theme provider stories before reintroducing it. https://jira.d2iq.com/browse/D2IQ-79209
-
```
Comment (nataliepina) [inline packages/themes/stories/UIKitThemeProvider.stories.tsx:2]: This has been resolved and we do display this story now.

**Review by vacas5 — APPROVED** [review]

**Pull request merged**